### PR TITLE
Docs: signatures for methods returning relations

### PR DESCRIPTION
**Pull request by jaredbeck**
I couldn't find any docs on this, so I'm attempting to write some. I'm not at all sure this is correct, but will at least start a conversation. Thanks!

### CODE REVIEW

**Comment by manhhung741** [comment]
Jared, we do support using Bacon::ActiveRecord_Relation. I do agree with
need better documentation for it

**Comment by jaredbeck** [comment]
> we do support using Bacon::ActiveRecord_Relation ..

Oh, great! Should I update this PR to make that the recommendation?

**Comment by manhhung741** [comment]
Sure, I think we need it. Hopefully we can resolve the issue like with #211 

Thanks for helping with the documentation. Here is some thought I've had:
- We need a whole section calling out things `sorbet-rails` provide at runtime (as opposed to things provided via the static generator). It includes following functionality:
+ Relation class: Making the relations available at runtime (they are private, the gem make them public)
+ Model: find_n, first_n, last_n are the methods we added. pluck_to_tstruct as well.
+ Controller: adding fetch_typed and require_typed

- And make sure people put `sorbet-rails` in a common group in their Gemfile (not development or test only)

**Comment by jaredbeck** [comment]
> Here is some thought I've had

Nice! Thanks Harry. I've added your thoughts, and re-organized a little.

File: README.md [hunk 1]
```diff
@@ -19,16 +19,14 @@ This gem adds a few Rake tasks to generate Ruby Interface (RBI) files for dynami
 
 ```
 # -- Gemfile --
-
+# In the default group (same as sorbet-runtime) because of "Features Provided at Runtime" below
```
Comment (manhhung741) [inline README.md]: I've heard some people don't use `sorbet-runtime` in the production environment, so I think this part may actually be misleading. Maybe we should say "the group including production environment"

File: README.md [hunk 2]
```diff
@@ -148,6 +146,22 @@ Since mailing action methods is based on instance methods defined in a mailer cl
 - If not, all the params in the mailing action method will be T.untyped.
 - For return type though, the mailing action method will return `ActionMailer::MessageDelivery` instead of the return type of the instance method.
 
+## Features Provided at Runtime
```
Comment (manhhung741) [inline README.md]: I think we should change the current "Type-checking Rails code" to "Static RBI Generation", and move the documentation of run-time features there to this section.

Comment (manhhung741) [reply README.md]: What's needed for this section to work properly is running `rake rails_rbi:custom`, which will copy the rbi for the provided methods.

**Comment by jaredbeck** [comment]
Updated per comments, please review.

> What's needed for this section to work properly is running rake rails_rbi:custom, which will copy the rbi for the provided methods.

I don't understand this comment. Is it something you can add in a future PR?

File: README.md [hunk 3]
```diff
@@ -148,6 +147,23 @@ Since mailing action methods is based on instance methods defined in a mailer cl
 - If not, all the params in the mailing action method will be T.untyped.
 - For return type though, the mailing action method will return `ActionMailer::MessageDelivery` instead of the return type of the instance method.
 
+### Features Provided at Runtime
```
Comment (manhhung741) [inline README.md]:
```suggestion
## Runtime Features
```

**Comment by manhhung741** [comment]
Btw, please remove `[ci skip]` from your commit message. Github checks the PR passes the Travis CI check before it can be merged. Our repo is set up to requires PR passes CI run to merge. Even though this change doesn't make any code change, it's easier to have consistent rule on this.

**Comment by manhhung741** [comment]
> I don't understand this comment. Is it something you can add in a future PR?
Sure, I'll add it later. I'll attempt to explain it still: simply put we need to copy some RBI so that the sorbet knows the signature of the runtime methods added. It's through running the rake task
```
rake rails_rbi:custom
```

**Comment by jaredbeck** [comment]
Thanks Harry, changes made.

**Pull request merged**